src/components/expression/expression-renderer.js

```diff
@@ -594,7 +633,119 @@ genotet.ExpressionRenderer.prototype.drawMatrixCells_ = function() {
       this.signal('expressionClick', this.clickedObject_);
     }.bind(this));
   heatmapRects.exit().remove();
-  this.highlightLabelsAfterUpdateData_();
+
+  // Zoom selection.
+  var cellWidth = this.cellWidth;
+  var cellHeight = this.cellHeight;
+  var columnStart, columnEnd, rowStart, rowEnd;
+  var selected = false;
+
+  var zoomSelection = this.svgHeatmapContent_
+    .on('mousedown', function() {
+      var mousePosition = d3.mouse(this);
+      zoomSelection.append('rect')
+        .classed('selection', true)
+        .attr({
+          x: mousePosition[0],
+          y: mousePosition[1],
+          width: 1,
+          height: 1
+        });
+    })
+    .on('mousemove', function() {
+      var rectSelection = zoomSelection.select('rect.selection');
+      if (!rectSelection.empty()) {
+        var mousePosition = d3.mouse(this);
+        var selectedRange = {
+          height: parseInt(rectSelection.attr('height'), 10),
+          width: parseInt(rectSelection.attr('width'), 10),
+          x: parseInt(rectSelection.attr('x'), 10),
+          y: parseInt(rectSelection.attr('y'), 10)
+        };
+        var selectedBorderWeight = {
+          x: mousePosition[0] - selectedRange.x,
+          y: mousePosition[1] - selectedRange.y
+        };
+
+        if (selectedBorderWeight.x < 1 ||
+          (selectedBorderWeight.x * 2 < selectedRange.width)) {
+          selectedRange.x = mousePosition[0];
+          selectedRange.width -= selectedBorderWeight.x;
+        } else {
+          selectedRange.width = selectedBorderWeight.x;
+        }
+
+        if (selectedBorderWeight.y < 1 ||
+          (selectedBorderWeight.y * 2 < selectedRange.height)) {
+          selectedRange.y = mousePosition[1];
+          selectedRange.height -= selectedBorderWeight.y;
+        } else {
+          selectedRange.height = selectedBorderWeight.y;
+        }
+        rectSelection.attr(selectedRange);
+
+        d3.selectAll('.cell-selected').classed('cell-selected', false);
+        d3.selectAll('.label-selected').classed('label-selected', false);
+
+        rowStart = Math.floor(selectedRange.y / cellHeight);
+        rowEnd = Math.floor((selectedRange.y + selectedRange.height) /
+          cellHeight);
+        columnStart = Math.floor(selectedRange.x / cellWidth);
+        columnEnd = Math.floor((selectedRange.x + selectedRange.width) /
+          cellWidth);
+        for (var i = rowStart; i <= rowEnd; i++) {
+          d3.select('.gene-labels text:nth-of-type(' + (i + 1) + ')')
```

Comment:
Seems dangerous and inefficient. How do you ensure the order of the d3 appended elements so that nth-of-type always works properly?